Cherry-pick update libraries

terrakok opened the pull request:
## Release Notes
https://github.com/JetBrains/compose-multiplatform-core/pull/2075
https://github.com/JetBrains/compose-multiplatform-core/pull/2078
https://github.com/JetBrains/compose-multiplatform-core/pull/2079
https://github.com/JetBrains/compose-multiplatform-core/pull/2093

terrakok committed:
Savedstate 1.3.0 (#2075)

## Release Notes
N/A

terrakok committed:
Lifecycle 2.9.0 (#2078)

## Release Notes
N/A

terrakok committed:
Fix navigator registration with the correct name (#2093)

Previously, the navigator's name property was being used instead of the
provided `name` argument, which could lead to a null exception.

## Release Notes
N/A

terrakok committed:
Navigation 2.9.0 (#2079)

## Release Notes
N/A

igordmn commented:
~~CfW tests should be fixed first~~ Fixed

igordmn commented:
Should be merged as "Create merge commit" or "Rebase and merge", not as "Squash and merge"

terrakok committed:
Correct casing for `nonJvmMain` in build.gradle config.

Updated all occurrences of `nonjvmMain` to `nonJvmMain` to ensure consistent casing in module dependencies. This fixes build issues on linux CI agents with a case-sensitive file system.
